scene/2d/line_builder.cpp

```diff
@@ -30,91 +30,43 @@
 
 #include "line_builder.h"
 
-//----------------------------------------------------------------------------
-// Util
-//----------------------------------------------------------------------------
-
-enum SegmentIntersectionResult {
-	SEGMENT_PARALLEL = 0,
-	SEGMENT_NO_INTERSECT = 1,
-	SEGMENT_INTERSECT = 2
-};
-
-static SegmentIntersectionResult segment_intersection(
-		Vector2 a, Vector2 b, Vector2 c, Vector2 d,
-		Vector2 *out_intersection) {
-	// http://paulbourke.net/geometry/pointlineplane/ <-- Good stuff
-	Vector2 cd = d - c;
-	Vector2 ab = b - a;
-	float div = cd.y * ab.x - cd.x * ab.y;
-
-	if (Math::abs(div) > 0.001f) {
-		float ua = (cd.x * (a.y - c.y) - cd.y * (a.x - c.x)) / div;
-		float ub = (ab.x * (a.y - c.y) - ab.y * (a.x - c.x)) / div;
-		*out_intersection = a + ua * ab;
-		if (ua >= 0.f && ua <= 1.f &&
-				ub >= 0.f && ub <= 1.f) {
-			return SEGMENT_INTERSECT;
-		}
-		return SEGMENT_NO_INTERSECT;
-	}
-
-	return SEGMENT_PARALLEL;
-}
-
-static float calculate_total_distance(const Vector<Vector2> &points) {
-	float d = 0.f;
-	for (int i = 1; i < points.size(); ++i) {
-		d += points[i].distance_to(points[i - 1]);
-	}
-	return d;
-}
-
-static inline Vector2 rotate90(const Vector2 &v) {
-	// Note: the 2D referential is X-right, Y-down
-	return Vector2(v.y, -v.x);
-}
+#include "core/math/geometry_2d.h"
 
+// Utility method.
 static inline Vector2 interpolate(const Rect2 &r, const Vector2 &v) {
 	return Vector2(
 			Math::lerp(r.position.x, r.position.x + r.get_size().x, v.x),
 			Math::lerp(r.position.y, r.position.y + r.get_size().y, v.y));
 }
 
-//----------------------------------------------------------------------------
-// LineBuilder
-//----------------------------------------------------------------------------
-
 LineBuilder::LineBuilder() {
 }
 
-void LineBuilder::clear_output() {
-	vertices.clear();
-	colors.clear();
-	indices.clear();
-	uvs.clear();
-}
-
 void LineBuilder::build() {
-	// Need at least 2 points to draw a line
+	// Need at least 2 points to draw a line, so clear the output and return.
 	if (points.size() < 2) {
-		clear_output();
+		vertices.clear();
+		colors.clear();
+		indices.clear();
+		uvs.clear();
 		return;
 	}
 
 	ERR_FAIL_COND(tile_aspect <= 0.f);
 
-	const float hw = width / 2.f;
-	const float hw_sq = hw * hw;
-	const float sharp_limit_sq = sharp_limit * sharp_limit;
-	const int len = points.size();
+	float hw = width / 2.f;
+	float modified_hw = hw;
+	float hw_sq = hw * hw;
+	float sharp_limit_sq = sharp_limit * sharp_limit;
+	int point_count = points.size();
+	bool wrap_around = closed && point_count > 2;
```

Comment:
I'd move the mutable variables down below with the other mutable ones and restore the `const` to the ones that don't need to be mutable, to communicate intent here, as the code already had `const`. Changing it doesn't make sense to me as it is correct and communicates intent